counterparty-core/counterpartycore/lib/messages/utxo.py

```diff
@@ -176,87 +264,62 @@ def parse(db, tx, message):
         status = "invalid: " + "; ".join(problems)
 
-    # prepare bindings
-    bindings = {
-        "tx_index": tx["tx_index"],
-        "tx_hash": tx["tx_hash"],
-        "msg_index": ledger.get_send_msg_index(db, tx["tx_hash"]),
-        "block_index": tx["block_index"],
-        "status": status,
-    }
-
     if status == "valid":
-        # fee payment only for attach to utxo
-        if action == "attach to utxo":
-            fee = gas.get_transaction_fee(db, ID, tx["block_index"])
-        else:
-            fee = 0
-        if fee > 0:
-            # fee is always paid by the address
-            if action == "attach to utxo":
-                fee_payer = source
-            else:
-                fee_payer = recipient
-            # debit fee from the fee payer
-            ledger.debit(
-                db,
-                fee_payer,
-                config.XCP,
-                fee,
-                tx["tx_index"],
-                action=f"{action} fee",
-                event=tx["tx_hash"],
-            )
-            # destroy fee
-            destroy_bindings = {
-                "tx_index": tx["tx_index"],
-                "tx_hash": tx["tx_hash"],
-                "block_index": tx["block_index"],
-                "source": tx["source"],
-                "asset": config.XCP,
-                "quantity": fee,
-                "tag": f"{action} fee",
-                "status": "valid",
-            }
-            ledger.insert_record(db, "destructions", destroy_bindings, "ASSET_DESTRUCTION")
-        # debit asset from source and credit to recipient
-        ledger.debit(
-            db, source, asset, quantity, tx["tx_index"], action=action, event=tx["tx_hash"]
-        )
-        ledger.credit(
-            db,
-            recipient,
-            asset,
-            quantity,
-            tx["tx_index"],
-            action=action,
-            event=tx["tx_hash"],
-        )
-        # we store parameters only if the transaction is valid
-        bindings = bindings | {
-            "source": source,
-            "destination": recipient,
-            "asset": asset,
-            "quantity": quantity,
-            "fee_paid": fee,
-        }
+        fee_paid = pay_fee(db, tx, action, source, recipient)
+
         # update counter
         if action == "attach to utxo":
             gas.increment_counter(db, ID, tx["block_index"])
 
-    ledger.insert_record(db, "sends", bindings, event)
+        if action == "attach to utxo" or not util.enabled("spend_utxo_to_detach"):
+            proceed_send(db, tx, action, source, recipient, asset, quantity, event, fee_paid)
+
+        if action == "detach from utxo" and util.enabled("spend_utxo_to_detach"):
+            # we detach all the assets from the source UTXO
+            balances = ledger.get_utxo_balances(db, source)
+            for balance in balances:
+                if balance["quantity"] == 0:
+                    continue
+                proceed_send(
```

Comment:
call this something else? `move_asset()`?

Reply:
Done